### PR TITLE
📜 add anonymisation script

### PR DESCRIPTION
**Pull request by eliascodes**
related to #383 

### Changes
- Add anonymisation script
- Update docs
- Rename a var
- Update deps

### Testing Requirements
- [x] N/A
  - I already used this to update the `twine-api-anon-data` dataset

### Release Requirements
- N/A

### Manual Deployment
- N/A

### NOTE
This fails when an organisation has more than one CB_ADMIN user. This _shouldn't_ be the case, but currently is the case for two CBs 🙈. I had to fix this manually when updating `twine-api-anon-data`.

### CODE REVIEW

File: api/bin/anonymise_data.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  const users = await Users.get(client);
+
+  try {
+
+    await client.transaction(async (trx) => {
+      await Promise.all(users.map(async (user) => {
+        const org = await Organisations.fromUser(client, { where: { id: user.id } });
+        const roles = await Roles.fromUserWithOrg(client, { userId: user.id, organisationId: org.id });
+        const name = String(user.id);
```
Comment (astroash) [inline api/bin/anonymise_data.ts]: Can we came the username `User ${user.id}`

File: api/bin/anonymise_data.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        if (org === null) {
+          throw new Error(`User ${user.name} has no organisation`);
+        }
+
+        const email = isAdmin
+          ? `admin@${org.name.toLowerCase().replace(/ /g, '').replace('é', 'e').replace(/[:&'-_]/g, '')}.com`
+          : `${name}@example.com`;
+
+        const password = isAdmin
```
Comment (astroash) [inline api/bin/anonymise_data.ts]: Is there a reason we're not resetting all passwords?

File: api/bin/anonymise_data.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        const roles = await Roles.fromUserWithOrg(client, { userId: user.id, organisationId: org.id });
+        const name = String(user.id);
+        const isAdmin = roles.some((role => role === RoleEnum.CB_ADMIN));
+
+        if (org === null) {
+          throw new Error(`User ${user.name} has no organisation`);
+        }
+
+        const email = isAdmin
+          ? `admin@${org.name.toLowerCase().replace(/ /g, '').replace('é', 'e').replace(/[:&'-_]/g, '')}.com`
```
Comment (astroash) [inline api/bin/anonymise_data.ts]: This format has changed, can you update lastpass

**Comment by astroash** [comment]
@eliascodes can you run this script on the remote anonymised api?

**Comment by eliascodes** [comment]
I ran the script on my local, it worked (taking into account the note in the PR description). Didn't want to run it on the anon DB right now because people are using it to review the visitor app charts. Updated the emails though, to be of the form `userXXXX@example.com`

**Pull request merged**